qiita_db/meta_util.py

```diff
@@ -581,70 +605,106 @@ def update_resource_allocation_redis(active=True):
-
-                fig, axs = resource_allocation_plot(df, col_name)
-                titles = [0, 0]
-                images = [0, 0]
-
-                # Splitting 1 image plot into 2 separate for better layout.
-                for i, ax in enumerate(axs):
-                    titles[i] = ax.get_title()
-                    ax.set_title("")
-                    # new_fig, new_ax – copy with either only memory plot or
-                    # only time
-                    new_fig = plt.figure()
-                    new_ax = new_fig.add_subplot(111)
-                    line = ax.lines[0]
-                    new_ax.plot(line.get_xdata(), line.get_ydata(),
-                                linewidth=1, color='orange')
-                    handles, labels = ax.get_legend_handles_labels()
-                    for handle, label, scatter_data in zip(handles,
-                                                           labels,
-                                                           ax.collections):
-                        color = handle.get_facecolor()
-                        new_ax.scatter(scatter_data.get_offsets()[:, 0],
-                                       scatter_data.get_offsets()[:, 1],
-                                       s=scatter_data.get_sizes(), label=label,
-                                       color=color)
-
-                    new_ax.set_xscale('log')
-                    new_ax.set_yscale('log')
-                    new_ax.set_xlabel(ax.get_xlabel())
-                    new_ax.set_ylabel(ax.get_ylabel())
-                    new_ax.legend(loc='upper left')
-
-                    new_fig.tight_layout()
-                    plot = BytesIO()
-                    new_fig.savefig(plot, format='png')
-                    plot.seek(0)
-                    img = 'data:image/png;base64,' + quote(
-                          b64encode(plot.getvalue()).decode('ascii'))
-                    images[i] = img
-                    plt.close(new_fig)
-                plt.close(fig)
-
-                # SID, CID, col_name
-                values = [
-                    ("img_mem", images[0], r_client.set),
-                    ("img_time", images[1], r_client.set),
-                    ('time', time, r_client.set),
-                    ("title_mem", titles[0], r_client.set),
-                    ("title_time", titles[1], r_client.set)
-                ]
-
-                for k, v, f in values:
-                    redis_key = 'resources$#%s$#%s$#%s$#%s:%s' % (
-                                cname, sname, version, col_name, k)
-                    r_client.delete(redis_key)
-                    f(redis_key, v)
+                # column_name_str looks like col1*col2*col3, etc
+                for col_name_str in col_name_list:
+                    df_copy = df.copy()
+                    new_column = None
+                    col_name_split = col_name_str.split('*')
+                    df_copy.dropna(subset=col_name_split, inplace=True)
```

Comment:
I think you need the copy cause here you are changing the dataframe, maybe just do:
df_copy = df.dropna(subset=col_name_split)